src/olympia/users/utils.py

```diff
@@ -285,3 +288,108 @@ def upload_picture(user, picture):
         user.picture_path,
         set_modified_on=user.serializable_reference(),
     )
+
+
+def assert_socket_labs_settings_defined():
+    if not settings.SOCKET_LABS_TOKEN:
+        raise Exception('SOCKET_LABS_TOKEN is not defined')
+
+    if not settings.SOCKET_LABS_HOST:
+        raise Exception('SOCKET_LABS_HOST is not defined')
+
+    if not settings.SOCKET_LABS_SERVER_ID:
+        raise Exception('SOCKET_LABS_SERVER_ID is not defined')
+
+
+utils_log = olympia.core.logger.getLogger('z.users')
+
+
+def check_suppressed_email_confirmation(verification=None, page_size=5):
+    assert_socket_labs_settings_defined()
+
+    if not verification:
+        raise Exception('Verification is not defined')
+
+    email = verification.suppressed_email.email
+
+    current_count = 0
+    total = 0
+
+    code_snippet = str(verification.confirmation_code)[-5:]
+    path = f'servers/{settings.SOCKET_LABS_SERVER_ID}/reports/recipient-search/'
+
+    # socketlabs might set the queued time any time of day
+    # so we need to check to midnight, one day before the verification was created
+    # and to midnight of tomorrow
+    before = verification.created - datetime.timedelta(days=1)
+    start_date = datetime.datetime(
+        year=before.year,
+        month=before.month,
+        day=before.day,
+    )
+    end_date = datetime.datetime.now() + datetime.timedelta(days=1)
+    date_format = '%Y-%m-%d'
+
+    params = {
+        'toEmailAddress': email,
+        'startDate': start_date.strftime(date_format),
+        'endDate': end_date.strftime(date_format),
+        'pageNumber': 0,
+        'pageSize': page_size,
+        'sortField': 'queuedTime',
+        'sortDirection': 'dsc',
+    }
+
+    is_first_page = True
+
+    found_emails = []
+
+    while current_count < total or is_first_page:
+        if not is_first_page:
+            params['pageNumber'] = params['pageNumber'] + 1
+
+        url = (
+            urllib.parse.urljoin(settings.SOCKET_LABS_HOST, path)
+            + '?'
+            + urllib.parse.urlencode(params)
+        )
+
+        headers = {
+            'authorization': f'Bearer {settings.SOCKET_LABS_TOKEN}',
+        }
+
+        utils_log.info(f'checking for {code_snippet} with params {params}')
+
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+        json = response.json()
```

Comment:
`json` here is overriding a built in python library 😬

Reply:
Odd that this is possible to commit. Vscode is also happy to see this.